lib/src/revset.pest

```diff
@@ -12,10 +12,10 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-non_period_identifier = @{ (ASCII_ALPHANUMERIC | "_" | "@" | "/")+ }
+non_final_identifier = @{ (ASCII_ALPHANUMERIC | "_" | "@" | "/")+ }
 identifier = @{
-  (non_period_identifier+ ~ ".")+ ~ non_period_identifier+
-  | non_period_identifier+
+  (non_final_identifier+ ~ ("." | "-" | "+" ))+ ~ non_final_identifier+
```

Comment:
How will this handle 'foo--bar', 'foo++bar', 'foo+-bar', etc.?  I don't know that they should be rejected as valid identifiers, but if so they should be tested.

Reply:
I meant to reject them, yes. That leaves the door open for using e.g. `++` as an infix operator in the future, and it's easier to relax it later than to reverse it if we relax it now. Good point about testing, I've added a few.